docs/references/t-ecdsa-how-it-works.mdx

```diff
@@ -1,73 +1,110 @@
+The availability of a particular `key_id` depends on implementation.<br/>
+For curve `secp256k1`, the public key is derived using a generalization of BIP32 (see [ia.cr/2021/1330](https://eprint.iacr.org/2021/1330), Appendix D). 
+To derive (non-hardened) BIP-0032-compatible public keys, each byte string (blob) in the `derivation_path` must be a 4-byte big-endian encoding of an unsigned integer less than 2<sup>31</sup>.<br/>
 The return result is an extended public key consisting of an ECDSA `public_key`, encoded in SEC1 compressed form, and a `chain_code`, which can be used to deterministically derive child keys of the `public_key`.
 This call requires that the ECDSA feature is enabled, and the `canister_id` meets the requirement of a canister id. Otherwise it will be rejected.
 -   [`sign_with_ecdsa`](/docs/current/references/ic-interface-spec#ic-ecdsa_public_key): this method returns a new ECDSA signature of the given `message_hash` that can be separately verified against a derived ECDSA public key. This public key can be obtained by calling `ecdsa_public_key` with the caller's `canister_id`, and the same `derivation_path` and `key_id` used here.<br/>
 The signatures are encoded as the concatenation of the SEC1 encodings of the two values `r` and `s`. For curve `secp256k1`, this corresponds to 32-byte big-endian encoding.<br/>
 This call requires that the ECDSA feature is enabled, the caller is a canister, and `message_hash` is 32 bytes long. Otherwise it will be rejected.
 
-An example of the API can be found below:
+-   [`schnorr_public_key`](/docs/current/references/ic-interface-spec#ic-sign_with_schnorr): this method returns a Schnorr public key for the given canister using the given derivation path. 
+If the `canister_id` is unspecified, it will default to the canister id of the caller. 
+The `derivation_path` is a vector of variable length byte strings. 
+The `key_id` is a struct specifying both a curve and a name. 
+The availability of a particular `key_id` depends on implementation.<br/>
+For curve `bip340secp256k1`, the public key is derived using a generalization of BIP32 (see [ia.cr/2021/1330](https://eprint.iacr.org/2021/1330), Appendix D). 
+{/* Something about derivation and encoding */}
+This call requires that the Schnorr feature is enabled, and the `canister_id` meets the requirement of a canister id. Otherwise it will be rejected.
+-   [`sign_with_schnorr`](/docs/current/references/ic-interface-spec#ic-ecdsa_public_key): this method returns a new Schnorr signature of the given `message` that can be separately verified against a derived Schnorr public key. 
+This public key can be obtained by calling `schnorr_public_key` with the caller's `canister_id`, and the same `derivation_path` and `key_id` used here.<br/>
+{/* Something about encoding */}
+This call requires that the Schnorr feature is enabled, the caller is a canister, and `message` is 32 bytes long. Otherwise it will be rejected.
```

Comment:
```suggestion
-   [`schnorr_public_key`](/docs/current/references/ic-interface-spec#ic-sign_with_schnorr): Returns a Schnorr public key for the given canister using the given derivation path. 

If the `canister_id` is unspecified, it will default to the canister id of the caller. 
- The `derivation_path` is a vector of variable length byte strings. 
- The `key_id` is a struct specifying both a curve and a name. The availability of a particular `key_id` depends on implementation.

For curve `bip340secp256k1`, the public key is derived using a generalization of BIP32 (see [ia.cr/2021/1330](https://eprint.iacr.org/2021/1330), Appendix D). 

{/* Something about derivation and encoding */}

This call requires that the Schnorr feature is enabled, and the `canister_id` meets the requirement of a canister id. Otherwise it will be rejected.

-   [`sign_with_schnorr`](/docs/current/references/ic-interface-spec#ic-ecdsa_public_key): Returns a new Schnorr signature of the given `message` that can be separately verified against a derived Schnorr public key. 
This public key can be obtained by calling `schnorr_public_key` with the caller's `canister_id`, and the same `derivation_path` and `key_id` used here.

{/* Something about encoding */}
This call requires that the Schnorr feature is enabled, the caller is a canister, and `message` is 32 bytes long. Otherwise it will be rejected.
```